arm-compute/2016-09-30/swagger/containerService.json

```diff
@@ -0,0 +1,555 @@
+                "Standard_GS3",
+                "Standard_GS4",
+                "Standard_GS5"
+              ],
+              "x-ms-enum": {
+                "name": "ContainerServiceVMSizeTypes",
+                "modelAsString": true
+              },
+              "description": "Size of agent VMs"
+            },
+            "dnsPrefix": {
+              "type": "string",
+              "description": "DNS prefix to be used to create FQDN for this agent pool"
+            },
+            "fqdn": {
+              "readOnly": true,
+              "type": "string",
+              "description": "FDQN for the agent pool"
+            }
+          },
+          "required": [
+            "name",
+            "dnsPrefix"
+          ],
+          "description": "Profile for container service agent pool"
+        },
+        "ContainerServiceWindowsProfile": {
+          "properties": {
+            "adminUsername": {
+              "type": "string",
+              "description": "The administrator username to use for Windows VMs"
+            },
+            "adminPassword": {
+              "type": "string",
+              "description": "The administrator password to use for Windows VMs"
+            }
+          },
+          "required": [
+            "adminUsername",
+            "adminPassword"
+          ],
+          "description": "Profile for Windows VMs"
+        },
+        "ContainerServiceLinuxProfile": {
+          "properties": {
+            "adminUsername": {
+              "type": "string",
+              "description": "The administrator username to use for all Linux VMs"
+            },
+            "ssh": {
+              "$ref": "#/definitions/ContainerServiceSshConfiguration",
+              "description": "Specifies the ssh key configuration for Linux VMs"
+            }
+          },
+          "required": [
+            "adminUsername",
+            "ssh"
+          ],
+          "description": "Profile for Linux VMs"
+        },
+        "ContainerServiceSshConfiguration": {
+          "properties": {
+            "publicKeys": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ContainerServiceSshPublicKey"
+              },
+              "description": "the list of SSH public keys used to authenticate with Linux based VMs"
+            }
+          },
+          "description": "SSH configuration for Linux based VMs running on Azure"
+        },
+        "ContainerServiceSshPublicKey": {
+          "properties": {
+            "keyData": {
+              "type": "string",
+              "description": "Certificate public key used to authenticate with VM through SSH. The certificate must be in Pem format with or without headers."
+            }
+          },
+          "required": [
+            "keyData"
+          ],
+          "description": "Contains information about SSH certificate public key data."
+        },
+        "ContainerServiceDiagnosticsProfile": {
```

Comment:
If "ContainerServiceDiagnosticsProfile" is provided then should "vmDiagnostics" always be provided i.e. should this property be required ?